internal/apex/wg-iface.go

```diff
@@ -0,0 +1,195 @@
+package apex
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+// getInterfaceByIP will looks ip an interface by the IP provided
+func getInterfaceByIP(ip net.IP) (string, error) {
+	interfaces, err := net.Interfaces()
+	if err != nil {
+		return "", err
+	}
+	for _, iface := range interfaces {
+		if addrs, err := iface.Addrs(); err == nil {
+			for _, addr := range addrs {
+				if ifaceIP, _, err := net.ParseCIDR(addr.String()); err == nil {
+					if ifaceIP.Equal(ip) {
+						return iface.Name, nil
+					}
+				} else {
+					continue
+				}
+			}
+		} else {
+			continue
+		}
+	}
```

Comment:
This is just a style choice, but I find this nesting to be a bit deep. We can reduce it a bit by flipping the `err` checks like this. It also lets you drop the `else` clauses.

```suggestion
	for _, iface := range interfaces {
		if addrs, err := iface.Addrs(); err != nil {
			continue
		}
		for _, addr := range addrs {
			if ifaceIP, _, err := net.ParseCIDR(addr.String()); err != nil {
				continue
			}
			if ifaceIP.Equal(ip) {
				return iface.Name, nil
			}
		}
	}
```